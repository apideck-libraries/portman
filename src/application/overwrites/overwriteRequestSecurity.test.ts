import { overwriteRequestSecurity } from '.'
import { getPostmanMappedOperation } from '../../../__tests__/testUtils/getPostmanMappedOperation'

describe('overwriteRequestSecurity', () => {
  it('should overwrite the request apiKey definition', async () => {
    const overwriteValues = {
      apiKey: {
        key: 'x-apideck-app-id',
        value: 'foo-bar-baz'
      }
    }
    const pmOperation = await getPostmanMappedOperation()
    overwriteRequestSecurity(overwriteValues, pmOperation)
    expect(pmOperation.item.getAuth()).toMatchSnapshot()
  })

  it('should overwrite the request basic auth', async () => {
    const overwriteValues = {
      basic: {
        username: 'foo',
        password: 'bar'
      }
    }
    const pmOperation = await getPostmanMappedOperation()
    overwriteRequestSecurity(overwriteValues, pmOperation)
    expect(pmOperation.item.getAuth()).toMatchSnapshot()
  })

  it('should overwrite the request bearer token auth', async () => {
    const overwriteValues = {
      bearer: {
        token: 'foo'
      }
    }
    const pmOperation = await getPostmanMappedOperation()
    overwriteRequestSecurity(overwriteValues, pmOperation)
    expect(pmOperation.item.getAuth()).toMatchSnapshot()
  })
})
