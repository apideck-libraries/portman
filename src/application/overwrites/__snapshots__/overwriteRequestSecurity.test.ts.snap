// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overwriteRequestSecurity should overwrite the request apiKey definition 1`] = `
Object {
  "apikey": Array [
    Object {
      "key": "key",
      "type": "any",
      "value": "x-apideck-app-id",
    },
    Object {
      "key": "value",
      "type": "any",
      "value": "foo-bar-baz",
    },
  ],
  "bearer": Array [
    Object {
      "key": "token",
      "type": "any",
      "value": "<Bearer Token>",
    },
  ],
  "type": "apikey",
}
`;

exports[`overwriteRequestSecurity should overwrite the request basic auth 1`] = `
Object {
  "basic": Array [
    Object {
      "key": "username",
      "type": "any",
      "value": "foo",
    },
    Object {
      "key": "password",
      "type": "any",
      "value": "bar",
    },
  ],
  "bearer": Array [
    Object {
      "key": "token",
      "type": "any",
      "value": "<Bearer Token>",
    },
  ],
  "type": "basic",
}
`;

exports[`overwriteRequestSecurity should overwrite the request bearer token auth 1`] = `
Object {
  "bearer": Array [
    Object {
      "key": "token",
      "type": "any",
      "value": "foo",
    },
  ],
  "type": "bearer",
}
`;
